src/outputs/parthenon_hdf5.cpp

```diff
@@ -493,14 +505,34 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
       HDF5WriteND(g_var, vname, host_data.data(), vinfo.tensor_rank + 1, local_offset,
                   local_count, global_count, pl_xfer, H5P_DEFAULT);
     }
+    std::vector<Real> pos_tmp; // tmp vector to (potentially) hold particle positions
     auto &rvars = std::get<SwarmInfo::MapToVarVec<Real>>(swinfo.vars);
     for (auto &[vname, swmvarvec] : rvars) {
       const auto &vinfo = swinfo.var_info.at(vname);
       auto host_data = swinfo.FillHostBuffer(vname, swmvarvec);
       SetCounts(swinfo, vinfo);
       HDF5WriteND(g_var, vname, host_data.data(), vinfo.tensor_rank + 1, local_offset,
                   local_count, global_count, pl_xfer, H5P_DEFAULT);
+      if (output_params.write_swarm_xdmf &&
+          (vname == swarm_position::x::name() || vname == swarm_position::y::name() ||
+           vname == swarm_position::z::name())) {
+        pos_tmp.insert(pos_tmp.end(), host_data.begin(), host_data.end());
+      }
     }
+    if (output_params.write_swarm_xdmf) {
+      const int npart = pos_tmp.size() / 3;
+      std::vector<Real> swarm_positions(pos_tmp.size());
+      int spcnt = 0;
+      for (int i = 0; i < npart; ++i) {
+        swarm_positions[spcnt++] = pos_tmp[i];
+        swarm_positions[spcnt++] = pos_tmp[npart + i];
+        swarm_positions[spcnt++] = pos_tmp[2 * npart + i];
+      }
+      SetCountsParticlePositions(swinfo);
+      HDF5WriteND(g_var, "swarm_positions", swarm_positions.data(), 2, local_offset,
+                  local_count, global_count, pl_xfer, H5P_DEFAULT);
```

Comment:
This implies (I think) that we're writing swarm positions twice, right? Since you're outputting the fused version for xdmf but still outputting the original in whatever requested shape.

Is that desirable? Or should we change swarms to always be read and written in this way? As a conceit for visualization I think I'm fine doing it this way as the original shape is better IMO and more faithful to parthenon. Could go either way.

Reply:
That is correct.  This was an intentional choice; but I equally hate it.  Here was my thought process:

(1) As you mention, this is totally wasteful; I am dumping the exact same info twice; just one in a particular format to make XDMF happy.  
(2) If I just switch to the format that XDMF requires, then (a) this would be a breaking change; (b) I would subsequently have to go in and make edits to fix swarm restarts; (c) I would have to make changes to `phdf.py`.
(3) In the spirit of trying be non-breaking, I keep restarts and dumps as is, just adding this new geometry piece iff swarm XDMF files are requested (which weren't available before).

Reply:
We should probably get other people's opinion on this... but I think I'm fine with your approach. Doing it the other way would require changes to both `phdf.py` and the restart reader. And I prefer the original layout (obviously... as I designed it.)

Perhaps a question is how does OpenPMD want it?

Reply:
In my most recent commit, I have opted to make `write_swarm_xdmf` `false` by default.  I feel this better reflects the discussion here; i.e., if you _really_ want to read particles into VisIt, then you incur the penalty of dumping geometry twice and creating an additional XDMF file.  Furthermore, the default behavior just recovers what is the current behavior in `develop` prior to this MR (but with the field variable XDMF fix).

Reply:
I'm fine with this.